File: subprojects/docs/src/samples/userguide/plugins/mutableManagedProperty/java/buildSrc/src/main/java/UrlProcess.java
import org.gradle.api.DefaultTask;
import org.gradle.api.tasks.TaskAction;

import java.net.URI;

public abstract class UrlProcess extends DefaultTask {
    // Use an abstract getter and setter method
    abstract URI getUri();
    abstract void setUri(URI uri);

    @TaskAction
    void run() {
        // Use the `uri` property
        System.out.println("Downloading " + getUri());
    }
}

File: subprojects/docs/src/samples/userguide/plugins/mutableManagedProperty/kotlin/buildSrc/src/main/kotlin/UrlProcess.kt

import org.gradle.api.DefaultTask
import org.gradle.api.tasks.TaskAction
import java.net.URI

abstract class UrlProcess : DefaultTask() {
    // Use an abstract var
    abstract var uri: URI

    @TaskAction
    fun run() {
        // Use the `uri` property
        println("Downloading $uri")
    }
}
